app/modules/users/parameters.py

```diff
@@ -241,6 +241,24 @@ def replace(cls, obj, field, value, state):
         if field == User.profile_fileupload_guid.key:
             value = cls.add_replace_profile_fileupload(value)
 
+        if field == User.email.key:
+            # Reuse the marshmallow email validator to check for validity
+            from marshmallow.exceptions import ValidationError
+            from marshmallow.validate import Email as EmailValidator
+
+            try:
+                EmailValidator()(value)
+            except ValidationError as ex:
+                abort(code=HTTPStatus.UNPROCESSABLE_ENTITY, message=str(ex))
+            user = User.query.filter_by(email=value).first()
+            if user and user != current_user:
+                abort(
+                    code=HTTPStatus.UNPROCESSABLE_ENTITY,
+                    message='Email address already in use',
```

Comment:
i suspect this would have still "worked" (blown up spectacularly) upon persisting, due to email being unique in the db, but i do prefer your explicit catching of this first.